### ISSUE TITLE
Two issues - Export-M365DSCConfiguration with credential authentication parameter either still prompts for credential or uses cached credential 

### ISSUE THREAD

**Issue by KuotingChiu** [microsoft/Microsoft365DSC#2255]
Export-M365DSCConfiguration -Path $Path -FileName $FileName -Workload $Workload -mode Default -Credential $Credential

With credential authentication passing in the $credential parameter, either UI still prompts for credential or it will used the cached credential.    To make this issue worse, we are using this for multiple tenant, so passing the $credential parameter with tenant A may result in using the tenant B cached credential and thus exporting tenant B configuration instead

**Comment by andikrueger** [comment]
What version of M365DSC are you using and could you run an update to the latest version:

```powershell
Update-Module Microsoft365DSC
Update-M365DSCDependencies
Uninstall-M365DSCOutdatedDependencies
```

**Comment by NikCharlebois** [comment]
We are clearing the cached credential on every run of Export-M365DSCConfiguration so this scenario should never happen in theory:
![image](https://user-images.githubusercontent.com/2547149/191621864-d97f6eb8-ea1c-4bec-b6a8-0235faaa4e27.png)

**Comment by NikCharlebois** [comment]
@KuotingChiu what workload is prompting for credential? We recently fixed an issue with an EXO resource that was indeed causing a re-prompt. As @andikrueger suggests, try updating to the latest version. Thanks

**Comment by KuotingChiu** [comment]
I was using the latest version and that time the version was 1.22.831.1 .  I can see now the latest version is 1.22.921.1 that I can re-test again.   AAD, EXO, OD, PP, and SPO all prompt for credential.

**Comment by KuotingChiu** [comment]
Confirmed with 1.22.921 version, at least SPO workload still prompt for credential, other workloads still need to be tested.

**Comment by andikrueger** [comment]
Are you able to connect via PnP Management without any prompt:

$Cred = Get-Credential
Connect-PnPOnline -Url <URL> -Credential $Cred

**Comment by KuotingChiu** [comment]
yes

> Are you able to connect via PnP Management without any prompt:
> 
> $Cred = Get-Credential Connect-PnPOnline -Url -Credential $Cred

yes, i tried that already without getting prompted.

**Comment by andikrueger** [comment]
Could you run the following test:

```powershell
Connect-M365Tenant -Workload PnP -Credential $cred -Url <SharePointAdminUrl>
```

**Comment by KuotingChiu** [comment]
Sorry Andi,  

Connect-PnPOnline returns with the following error and Connect-M365Tenant prompts for credential

Connect-PnPOnline : AADSTS50076: Due to a configuration change made by your administrator, or because you moved to a new location, you must use multi-factor authentication to
access '00000003-0000-0ff1-ce00-000000000000'.
Trace ID: dd2c858d-aa45-4ed9-a19d-0b5bd8420e00
Correlation ID: d5e59b96-3f55-4b69-9fc8-f7c5d37ba840
Timestamp: 2022-09-27 00:04:14Z
At line:1 char:1

**Comment by KuotingChiu** [comment]
It seems the account was getting flagged as a risky user then due to a changed location.   We use Zscaler as the cloud proxy for URLs configured in a PAC file.  This issue maybe related to the URL behind the scene for Connect-pnPOnline is not in our PAC file.  Is Connet-M365Tenant the same ?

**Comment by andikrueger** [comment]
It is the same. MSCloudlogin aka. Connect-M365Tenant has some additional features that we use with M365DSC. The connection gets established by connect-PnPOnline. 

Please investigate this issue on your side.  This is not related to m365dsc in this case. 

I’ll close this issue for now.

**Comment by KuotingChiu** [comment]
I took a look again, MFA definitely got triggered from what I can see in the Azure sign in log but there is really no identity protection or conditional access in place causing it.  https://learn.microsoft.com/en-us/answers/questions/836429/aadsts50076-due-to-a-configuration-change-made-by.html  suggests the following "It turned out that sometime in the past year the tenant we were operating in turned on an "Acceptable Use" conditional access policy. Although we had granted tenant wide granted admin consent, our application layer was failing to authenticate non-interactively on behalf of the user when making calls to Graph due to this policy.

Turns out, after a long conversation with Microsoft, there is no way to bypass this conditional access policy with admin consent, it must be interactive.

The (less than ideal) solution was to convert the required permissions to application rather than delegated. "

**Comment by andikrueger** [comment]
Thanks for the feedback. This is good to know.

**Comment by KuotingChiu** [comment]
I opened a formal Microsoft incident Case #:33339250 , currently being assigned to AAD support group on understanding why and how MFA got triggered (while nothing found in either conditional access, identity protection, or security default) and any means to disable it.   With MFA in place, automation is not possible with the delegated type of permissions

**Comment by andikrueger** [comment]
Did you enable AAD security defaults?

**Comment by KuotingChiu** [comment]
No, security default is disabled and Identity protections (sign in and user risks) are disabled as well.  There are no conditional access policies will trigger MFA.

**Comment by KuotingChiu** [comment]
After further review.  I found the per-user MFA is set on my admin ID that caused the error of "AADSTS50076: Due to a configuration change made by your administrator, or because you moved to a new location, you must use multi-factor authentication".  After removing the per-user MFA.  Now I am getting a different error with "Connect-PnPOnline -Url -Credential $Cred".   The new error is "Sign-in error code 50126, Failure reason Error validating credentials due to invalid username or password."  This error is bogus as the ID/Password are definitely correct.    I researched a little bit more and the -credential method of Connect-PnPOnline seems to use the basic authentication method and caused this error.   We have a conditional access policy to block basic authentication that seems to further contribute to this issue.   Are you aware of this issue?
